File: 11_replace_spaces_percent20.py
"""
Task 11: Replace all spaces in a string with %20.
Reads a line and prints the modified string.
"""
import sys

def replace_spaces(s: str) -> str:
    return ''.join('%20' if ch == ' ' else ch for ch in s)

if __name__ == '__main__':
    s = sys.stdin.readline().rstrip('\n')
    print(replace_spaces(s))

# Time Complexity: O(n)
# Space Complexity: O(n)

File: 11_replace_spaces_percent20.cpp
#include <bits/stdc++.h>
using namespace std;

/*
Task 11: Replace all spaces in a string with %20.
Reads a line and prints the modified string.
*/

int main()
{
    string s;
    if (!getline(cin, s))
        return 0;
    for (char c : s)
    {
        if (c == ' ')
            cout << "%20";
        else
            cout << c;
    }
    cout << '\n';
    return 0;
}

// Time Complexity: O(n)
// Space Complexity: O(1)
